internal/integration/kafka/kafka.go

```diff
@@ -52,10 +53,24 @@ func New(m marshaler.Type, conf config.IntegrationKafkaConfig) (*Integration, er
 	}
 
 	if conf.Username != "" || conf.Password != "" {
-		wc.Dialer.SASLMechanism = plain.Mechanism{
-			Username: conf.Username,
-			Password: conf.Password,
+		switch conf.Mechanism {
+		case "plain":
+			wc.Dialer.SASLMechanism = plain.Mechanism{
+				Username: conf.Username,
+				Password: conf.Password,
+			}
+		case "scram":
+			algo := scram.SHA512
```

Comment:
Would it be an idea to add a `switch` for the `conf.Algorithm` with two cases to handle `SHA-512` and `SHA-256` and a `default` returning an error that an invalid algorithm was configured? Just to make sure that the user gets an error in case of a typo.